sympy/geometry/ellipse.py

```diff
@@ -1005,6 +1005,45 @@ def semilatus_rectum(self):
         """
         return self.major * (1 - self.eccentricity ** 2)
 
+    def auxiliary_circle(self):
+        """Returns equation of auxiliary circle of the ellipse.
+
+        Auxiliary Circle of an ellipse is circumcircle of an ellipse i.e. the circle
+        whose center concurs with that of the ellipse and whose radius is equal to the ellipse's semi-major axis.
+
+        Parameters
+        ==========
+
+        x : str, optional
+            Label for the x-axis. Default value is 'x'.
+        y : str, optional
+            Label for the y-axis. Default value is 'y'.
+
+        Returns
+        =======
+
+        equation : sympy expression
```

Comment:
Looks like it actually returns a Circle object. Can you check how other geometry docstrings report their return types and make sure this is consistent.

Reply:
when I used the ellipse of the type having both `hradius` and `vradius` different so it gives
```
>>> from sympy import *
>>> e1 = Ellipse(Point2D(1, 5), 2, 3)
>>> e = e1.auxiliary_circle()
>>> e.equation()
(x - 1)**2 + (y - 5)**2 - 9
```
but in case of ellipse having `hradius`  = `vradius`  it gives
```
>>> from sympy import *
>>> e1 = Ellipse(Point2D(1, 5), 2, 2)
>>> e = e1.auxiliary_circle()
>>> e.equation()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-7-d07ce3f45221> in <module>()
----> 1 e.equation()

AttributeError: 'Add' object has no attribute 'equation'
```
but when I used this
```
>>> from sympy import *
>>> e1 = Ellipse(Point2D(1, 5), 2, 2)
>>> e = e1.auxiliary_circle()
>>> e
(x - 1)**2 + (y - 5)**2 - 4
```
it gives the result but not with `e.equation()` this happen with circle object too i.e.
```
>>> from sympy import *
>>> e1 = Circle(Point2D(1, 5), 4)
>>> e = e1.auxiliary_circle()
>>> e.equation()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-11-d07ce3f45221> in <module>()
----> 1 e.equation()

AttributeError: 'Add' object has no attribute 'equation'
>>> e
(x - 1)**2 + (y - 5)**2 - 16
```
why this happens I want this code to be consistent means it should return the auxiliary circle equation with single code `self.equation()` for all cases whether it is any ellipse or circle

Reply:
ping @oscarbenjamin

Reply:
I just tested each of those examples with this PR and they all seem to work fine. I didn't see the errors you report and in each the result returns by `e1.auxiliary_circle` was a `Circle` object.

Reply:
@oscarbenjamin but `e.equation()` does not work for a circle it does not return the auxiliary circle is it returning for your case?

Reply:
I don't understand the issue you are describing here. `e.equation` works for a circle as far as I can tell:
```julia
In [1]: Circle((0, 0), 1)                                                                                                                      
Out[1]: Circle(Point2D(0, 0), 1)

In [2]: Circle((0, 0), 1).equation()                                                                                                           
Out[2]: 
 2    2    
x  + y  - 1
```

Reply:
I am talking of this whole code, I had run this on jupyter notebook
```
>>> from sympy import *
>>> e1=Circle((0, 0), 1)
>>> e=e1.auxiliary_circle()
>>> e.equation()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-22-de93a72cf360> in <module>()
      2 e1=Circle((0, 0), 1)
      3 e=e1.auxiliary_circle()
----> 4 e.equation()

AttributeError: 'Add' object has no attribute 'equation'
```
above does not work for circle and ellipse having `hradius=vradius` though both are same 
it only returns auxiliary cirlce when I used below code
```
>>> from sympy import *
>>> e1=Circle((0, 0), 1)
>>> e=e1.auxiliary_circle()
>>> e
x**2 + y**2 - 1
```

Reply:
ping @oscarbenjamin

Reply:
Is it working for you now? I just checked out this PR and tried your code above and it seems to work as expected:
```julia
In [1]: e1=Circle((0, 0), 1)                                                                                                                   

In [2]: e=e1.auxiliary_circle()                                                                                                                

In [3]: e.equation()                                                                                                                           
Out[3]: 
 2    2    
x  + y  - 1

In [4]: e                                                                                                                                      
Out[4]: Circle(Point2D(0, 0), 1)
```

Reply:
Please don't ping me like this. I receive all email updates anyway. Apparently that ping arrived while I was writing my comment above. If I haven't responded yet then unless it's been at least a week pinging won't make me respond any faster.

Reply:
sorry for ping and inconvenience  @oscarbenjamin but no this is not  working on jupyter notebook where are you running this code and how
![111](https://user-images.githubusercontent.com/42106787/53287791-549aad80-37a7-11e9-8c3e-0d5b436f726f.JPG)

Reply:
I don't think you are running the code from this PR. Close jupyter etc and restart them. Then check the code that is actually in the files to see if it matches what is in this PR.

Reply:
Yes both have the same code since I have clone sympy repo in Anaconda 3 also and used the same code what is in PR in ellipse.py file of Anaconda 3 sympy repo and used Anaconda prompt to run jupyter notebook @oscarbenjamin